### PR TITLE
Feature multiple document esign

### PR DESCRIPTION
**Pull request by indraprasetya154**
## Overview
- Feature multiple document esign on mobile API
- I created some diagram, maybe it can be help reviewer 😸  https://whimsical.com/multi-file-tte-R7WBj8qnMpeUGbP2ioNTV3

## Example Mutator
````
mutation {
  documentSignatureMultiple(
    input: { documentSignatureSentIds: "8888, 9999", passphrase: "XXX" }
  ) {
    id
    status
    note
  }
}

````

## Related PR
- https://gitlab.com/jdsteam/sikd-website/-/merge_requests/528

## Related Task
- https://app.clickup.com/t/2y853tv

## Evidence
title: Feature multiple document esign on mobile API
project: SIKD
participants: @indraprasetya154 @samudra-ajri @fajarhikmal214 @azophy

### CODE REVIEW

File: src/database/migrations/2022_09_12_072623_create_failed_jobs_table.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        Schema::create('failed_jobs', function (Blueprint $table) {
+            $table->id();
+            $table->string('uuid')->unique();
+            $table->text('connection');
+            $table->text('queue');
+            $table->longText('payload');
+            $table->longText('exception');
+            $table->timestamp('failed_at')->useCurrent();
+        });
+    }
```
Comment (samudra-ajri) [inline src/database/migrations/2022_09_12_072623_create_failed_jobs_table.php:25]: Boleh tau table ini untuk apa, Kang?

Comment (indraprasetya154) [reply src/database/migrations/2022_09_12_072623_create_failed_jobs_table.php:25]: Ketika terdapat error di suatu jobs misal timeout, atau ada error dari code, maka state tersebut akan disimpan di table tersebut @samudra-ajri 
https://laravel.com/docs/8.x/queues#dealing-with-failed-jobs

Comment (samudra-ajri) [reply src/database/migrations/2022_09_12_072623_create_failed_jobs_table.php:25]: Klw sekedar disimpan di log belum cukup ya, Kang?

Comment (indraprasetya154) [reply src/database/migrations/2022_09_12_072623_create_failed_jobs_table.php:25]: hmmm bisa sih sebenernya, kita lempar ke kafka gitu ya?

Comment (samudra-ajri) [reply src/database/migrations/2022_09_12_072623_create_failed_jobs_table.php:25]: Ini karena lebutuhan dari BSRe tea bukan ya, Kang?

Comment (indraprasetya154) [reply src/database/migrations/2022_09_12_072623_create_failed_jobs_table.php:25]: bukan bukan, ini ga ada kaitannya sama yg bsre sih, ini lebih ke handle failed job multi tte

Comment (indraprasetya154) [reply src/database/migrations/2022_09_12_072623_create_failed_jobs_table.php:25]: keuntungannya kalau kita menggunakan metode table failed_job itu kalau ada job yang error bisa kita re run
https://laravel.com/docs/9.x/queues#retrying-failed-jobs @samudra-ajri

Comment (samudra-ajri) [reply src/database/migrations/2022_09_12_072623_create_failed_jobs_table.php:25]: Hho.. mantap, Kang.
Berarti memang ada kebutuhan lebih selain nyimpen log ya, Kang.👍

**Review by samudra-ajri — APPROVED** [review]

File: src/database/migrations/2022_09_12_072623_create_failed_jobs_table.php (hunk 1 above)
Comment (indraprasetya154) [reply src/database/migrations/2022_09_12_072623_create_failed_jobs_table.php:25]: yup betul betul, tapi itu bisa juga sih si failed job dimasukin ke kafka atau send notification ke mobile, mungkin next pr bisa kita sempurnain tergantung kebuthan kedepannya, makasih masukan dan diskusinya @samudra-ajri 🤝

**Review by fajarhikmal214 — APPROVED** [review]
I need to learn more about this next time, LGTM

File: src/app/Jobs/ProcessMultipleEsignDocument.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $userId                   = $this->userId;
+        $passphrase               = $this->passphrase;
```
Comment (azophy) [inline src/app/Jobs/ProcessMultipleEsignDocument.php:49]: future consideration: this field is pretty sensitive. We should think about better method to pass this data to queue safely. One of the easiest is to encrypt it on dispatch, and decrypt it here.

reference for built-in Laravel's Encryption : https://laravel.com/docs/9.x/encryption#encrypting-a-value

**Review by azophy — APPROVED** [review]

File: src/app/Jobs/ProcessMultipleEsignDocument.php (hunk 2 above)
Comment (indraprasetya154) [reply src/app/Jobs/ProcessMultipleEsignDocument.php:49]: next PR will be update it, thanks a lot mas @azophy

**Comment by azophy** [comment]
need to check further: make sure when queue job started, task status is set to processing

**Pull request merged**